### PR TITLE
Emailsender

### PR DESCRIPTION
**Pull request by Enanodr**
## Summary
- Request email with the latest ten posts functionality developed
- On the welcome page a link has been added for the email request
- Uses Mandrill as the email sender. 
- AppConfiguration gem used for managing sensitive data 

## TrelloCard

https://trello.com/c/ek4fgDQ9

### Link to request an email to the registered email
image

### Email with the latests 10 (or less if there aren't 10) articles
image

### CODE REVIEW

File: blog/Gemfile [hunk 1]
```diff
@@ -23,8 +23,17 @@ gem 'jbuilder', '~> 2.0'
 # bundle exec rake doc:rails generates the API under doc/api.
 gem 'sdoc', '~> 0.4.0', group: :doc
 
+gem 'omniauth'
```
Comment (epintos) [inline blog/Gemfile]: Its a good practice to leave a comment above the gems telling what they do

File: blog/app/assets/javascripts/comments.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
```
Comment (epintos) [inline blog/app/assets/javascripts/comments.coffee]: Remove unused files

File: blog/app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    redirect_to article_path(@article)
+  end
+
+  def destroy
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.find(params[:id])
+    @comment.destroy
+    redirect_to article_path(@article)
+  end
+
```
Comment (epintos) [inline blog/app/controllers/comments_controller.rb]: You can add memorization here

File: blog/app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  def create
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.create(comment_params)
+    redirect_to article_path(@article)
+  end
+
+  def destroy
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.find(params[:id])
+    @comment.destroy
```
Comment (epintos) [inline blog/app/controllers/comments_controller.rb]: It would be nice to add a success or error message

Comment (epintos) [inline blog/app/controllers/comments_controller.rb]: Same for the create

File: blog/app/controllers/omniauth_callbacks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  def facebook
+    generic_callback( 'facebook' )
+  end
+
+  def google_oauth2
+    generic_callback( 'google_oauth2' )
+  end
+
+  def generic_callback( provider )
+    @identity = Identity.find_for_oauth env["omniauth.auth"]
```
Comment (epintos) [inline blog/app/controllers/omniauth_callbacks_controller.rb]: You refactored this somehwere else right?

File: blog/app/helpers/comments_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
```
Comment (epintos) [inline blog/app/helpers/comments_helper.rb]: remove

File: blog/app/mailers/mailer.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class Mailer < ActionMailer::Base
+
+  def last_ten_posts_for_user(requesting_user)
+    @articles = Article.last(10)
+    mail subject: "Keep updated!", to: requesting_user.email, from: AppConfiguration.new.mail_user
```
Comment (epintos) [inline blog/app/mailers/mailer.rb]: You can also set in the Mailer above the from: email to avoid repeating it everywhere

**Comment by epintos** [comment]
I will continue lookiing at this when the diff is updated

File: blog/app/controllers/comments_controller.rb (hunk 3 above)
Comment (Enanodr) [inline blog/app/controllers/comments_controller.rb]: Refactored on father branch

**Comment by Enanodr** [comment]
Rebase from master, rubocop fixes were applied and fixes from the first review

**Comment by epintos** [comment]
Take into account my comment and :+1:

**Comment by epintos** [comment]
:+1:

**Pull request merged**